pepper-apis/src/main/java/org/broadinstitute/ddp/service/FileUploadService.java

```diff
@@ -117,6 +133,69 @@ URL generateSignedUrl(String blobName, String mimeType, HttpMethod method) {
                 Storage.SignUrlOption.withExtHeaders(headers));
     }
 
+    /**
+     * Determine if file meets criteria, and update its status appropriately. E.g. file must be uploaded, matches
+     * reported file size, and belong to the participant. Files already marked uploaded will not be checked again.
+     *
+     * @param handle            the database handle
+     * @param participantUserId the participant who this upload will be assigned to
+     * @param upload            the file upload
+     * @return check result
+     */
+    public CheckResult checkAndSetUploadStatus(Handle handle, long participantUserId, FileUpload upload) {
+        if (participantUserId != upload.getParticipantUserId()) {
+            return CheckResult.OWNER_MISMATCH;
```

Comment:
There's a few scenarios here, which we didn't discuss in API discussion, so I made some decisions here. One obvious one is that the file should "belong" to the participant. Another scenario:

* What should we do if the file is already "assigned"/"used" by another answer? Can we assign a file to multiple answers?
    * In this implementation, we're allowing it since it makes things easier and doesn't seem like an error.